service-tests/runner/pull-request-services-cli.js

```diff
@@ -18,6 +18,23 @@
 const difference = require('lodash.difference');
 const fetch = require('node-fetch');
 
+function inferPullRequest() {
+  if (process.env.TRAVIS) {
+    return {
+      repoSlug: process.env.TRAVIS_REPO_SLUG,
+      pullRequest: process.env.TRAVIS_PULL_REQUEST,
+    };
+  } else if (process.env.CIRCLECI) {
+    const prUrl = process.env.CI_PULL_REQUEST || '';
+    return {
+      repoSlug: prUrl.split('/').slice(-4, -2).join('/'),
```

Comment:
This can use `CIRCLE_PR_USERNAME` and `CIRCLE_PR_REPONAME`

Reply:
Experimentally that doesn't seem to be the case. In my own testing, I was getting my username in `CIRCLE_PR_USERNAME`, not `shields`. Also the docs say that these are only populated for forks, so they wouldn't work on Greenkeeper builds (or if we start developing branches within the shields repo).

Reply:
@paulmelnikow - Oh, that's unfortunate 😢  It's pretty confusing that the pull request URL would differ from the other environment variables, but it is what it is. 

I'm just worried this code is pretty fragile and would break if the URL format it's using ever changes (eg. what if a querystring argument is added?). If you can restrict the build to running on Node.js 7.0 or higher, you can use the URL class (https://nodejs.org/docs/latest/api/url.html#url_the_whatwg_url_api) to parse the URL. Alternatively, there's the legacy `url.parse` method (https://nodejs.org/docs/latest/api/url.html#url_legacy_url_api). I'd suggest parsing the URL and just using the `pathname`.

This is what we do in a bot for Babel:
```
exports.parsePullRequestURL = function(prURL: string): {
  owner: string,
  repo: string,
  pr: number
} {
    const path = url.parse(prURL).pathname;
    if (!path) {
      throw new Error('Invalid GitHub URL');
    }
    const parts = path.substr(1).split('/');
    if (parts.length !== 4 || parts[2] !== 'pull') {
      throw new Error(`Unexpected GitHub PR URL format: ${path}`);
    }

    const [owner, repo, _, pr] = parts;
    return {owner, repo, pr: +pr};
}
```
(https://github.com/babel/babel-bot/blob/master/src/github.js)

This has Flow typing but you could just strip that out and use the same logic.

Reply:
Cool, sounds good, I'll pull the URL path and add some checking.

Heh, I had some fun times recently with that URL class. 🙃